File: PHP/Patterns/src/Behavioral/Memento/TextEditor.php
<?php

namespace Behavioral\Memento;

class TextEditor
{
    private $content;

    public function __construct()
    {
        $this->content = '';
    }
    /**
     * @return string
     */
    public function getContent(): string
    {
        return $this->content;
    }

    /**
     * @return  void
     */
    public function setContent($content): void
    {
        $this->content = $content;
    }

    public function save(): TextEditorState
    {
        return new TextEditorState($this->content);
    }

    public function restore(TextEditorState $memento): void
    {
        $this->content = $memento->getContent();
    }
}

File: Python/momento/TextEditor.py
from TextEditorState import TextEditorState

class TextEditor:
    def __init__(self):
        self.__content = ""

    def getContent(self) -> str:
        return self.__content

    def setContent(self, content: str):
        self.__content = content

    def save(self) -> TextEditorState:
        return TextEditorState(self.__content)

    def restore(self, state: TextEditorState):
        self.__content = state.getContent()

